diff -crN orig/configure.ac fixed/configure.ac
*** orig/configure.ac	Mon Dec 29 08:32:43 2025
--- fixed/configure.ac	Mon Dec 29 10:00:45 2025
***************
*** 4071,4083 ****
     dnl libraries distributed are versioned 0.6.3.  We try to
     dnl accept a tree-sitter library that has incorrect version as long
     dnl as it supports changing malloc.
!   EMACS_CHECK_MODULES([TREE_SITTER], [tree-sitter >= 0.20.2],
      [HAVE_TREE_SITTER=yes], [HAVE_TREE_SITTER=no])
    if test "${HAVE_TREE_SITTER}" = yes; then
      AC_DEFINE(HAVE_TREE_SITTER, 1, [Define if using tree-sitter.])
      NEED_DYNLIB=yes
    else
!     EMACS_CHECK_MODULES([TREE_SITTER], [tree-sitter >= 0.6.3],
        [HAVE_TREE_SITTER=yes], [HAVE_TREE_SITTER=no])
      if test "${HAVE_TREE_SITTER}" = yes; then
        OLD_CFLAGS=$CFLAGS
--- 4071,4083 ----
     dnl libraries distributed are versioned 0.6.3.  We try to
     dnl accept a tree-sitter library that has incorrect version as long
     dnl as it supports changing malloc.
!   EMACS_CHECK_MODULES([TREE_SITTER], [tree-sitter >= 0.20.2 tree-sitter < 0.26],
      [HAVE_TREE_SITTER=yes], [HAVE_TREE_SITTER=no])
    if test "${HAVE_TREE_SITTER}" = yes; then
      AC_DEFINE(HAVE_TREE_SITTER, 1, [Define if using tree-sitter.])
      NEED_DYNLIB=yes
    else
!     EMACS_CHECK_MODULES([TREE_SITTER], [tree-sitter >= 0.6.3 tree-sitter < 0.26],
        [HAVE_TREE_SITTER=yes], [HAVE_TREE_SITTER=no])
      if test "${HAVE_TREE_SITTER}" = yes; then
        OLD_CFLAGS=$CFLAGS
